### PR TITLE
Em dashes instead of consecutive hyphens.

### PR DESCRIPTION
**Pull request by tfpf**
## PR Summary
Two consecutive hyphens were used to approximate an em dash when it wasn't available. Today, we have Unicode, and also an HTML character code for it. (The latter is, according to [this Stack Overflow answer](https://stackoverflow.com/a/52377269), the official method to insert em dashes.)

Also, it looks more professional than consecutive hyphens.

There are consecutive hyphens in multiple documentation files, but since the README file is directly presented to the user, it might be a good first place to make this change.

## PR Checklist
**Documentation and Tests**
- [N/A] Has pytest style unit tests (and `pytest` passes)
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by oscargus — APPROVED** [review]
I checked in my copy of The Chicago Manual of Style and, to my disappointment/surprise, there should indeed be no spaces around an em dash. 

It is not clear to me if using HTML-encoded characters in the thing to do in markdown though, so I'll not merge at this stage, but just confirm the space issue. Someone else that thinks HTML-encoding should go in markdown can merge though.

(I do not really know, just feels a bit "wrong", but then these is probably no "right" way of doing it either, using Unicode is most likely guaranteed to be messed up sooner or later...)

**Review by jklymak — COMMENTED** [review]
I'm not I favour of this.  Markdown is meant to be human readable if possible as a text file and this definitely reduces readability.  I'd just make these hyphens with spaces around them like the first sentence. Note this is informal and n dashes with spaces are fine in British English.

**Comment by oscargus** [comment]
If nothing else, it should be adjusted so that it is consistently using the same character(s) as em dash. Right now there is a single dash for the first and two for the remaining two.

I do not have a strong opinion on spaces or not. As said I expected that there should be spaces (however, I am not native in English, but I am quite sure that there are spaces in Swedish), but reading up that is the "correct" way. Which we of course can ignore.

File: README.md [hunk 1]
```diff
@@ -34,12 +34,11 @@ which is generated from `/doc/users/installing/index.rst`
 
 ## Contribute
 
-You've discovered a bug or something else you want to change -
-excellent!
+You've discovered a bug or something else you want to change&mdash;excellent!
```
Comment (oscargus) [inline README.md:37]:
```suggestion
You've discovered a bug or something else you want to change—excellent!
```

File: README.md [hunk 2]
```diff
@@ -34,12 +34,11 @@ which is generated from `/doc/users/installing/index.rst`
...
 
-You've worked out a way to fix it -- even better!
+You've worked out a way to fix it&mdash;even better!
```
Comment (oscargus) [inline README.md:39]:
```suggestion
You've worked out a way to fix it—even better!
```

File: README.md [hunk 3]
```diff
@@ -34,12 +34,11 @@ which is generated from `/doc/users/installing/index.rst`
...
 
-You want to tell us about it -- best of all!
+You want to tell us about it&mdash;best of all!
```
Comment (oscargus) [inline README.md:41]:
```suggestion
You want to tell us about it—best of all!
```

Comment (oscargus) [reply README.md:41]: These are all — although in a fixed-width font they look like -...

**Comment by tfpf** [comment]
Agreed that HTML escape sequences in a file which is supposed to be readable is not good. I have made the suggested changes … but will all browsers render Unicode correctly?

Happy to close this either way.

**Review by dstansby — COMMENTED** [review]
I'm pro having all of these as a single dash with a space on each side (` - `), which makes the markdown human readable.

**Comment by tfpf** [comment]
Will Unicode render correctly on all browsers?

**Review by dstansby — APPROVED** [review]

**Pull request merged**